### ISSUE TITLE
Provide $price to callback functions in edd_get_purchase_link

### ISSUE THREAD

**Issue by cklosowski** [easydigitaldownloads/Easy-Digital-Downloads#3849]
Initially proposed in pull request #3424 by @ericnicolaas. Waiting on new PR with clean history.

**Comment by pippinsplugins** [comment]
@cklosowski new PR ready for review.
